### PR TITLE
Add message actions

### PR DESCRIPTION
**Pull request by geekgonecrazy**

### CODE REVIEW

File: models/message.go [hunk 1]
```diff
@@ -73,3 +76,30 @@ type AttachmentField struct {
...
+
+type AttachmentActionType string
+
+const {
+	AttachmentActionTypeButton AttachmentActionType = "button"
+}
+
+// AttachmentAction are action buttons on message attachments
+type AttachmentAction struct {
+	Type AttachmentActionType `json:"type"`
```
Comment (graywolf336) [inline models/message.go]: What happened to the golang formatting on this one?

File: models/message.go [hunk 2]
```diff
@@ -61,6 +61,9 @@ type Attachment struct {
 
 	AudioURL string `json:"audio_url,omitempty"`
 	VideoURL string `json:"video_url,omitempty"`
+	
+	Actions []AttachmentAction `json:"actions,omitempty"`
```
Comment (graywolf336) [inline models/message.go]: Does `omitempty` work on slices?

Comment (geekgonecrazy) [reply models/message.go]: it should.  It should just omit the entire slice if its empty.

File: models/message.go (hunk 1 above)
Comment (geekgonecrazy) [reply models/message.go]: I did it in the browser 🙈

**Pull request merged**